cachi2/core/package_managers/yarn_classic/workspaces.py

```diff
@@ -0,0 +1,23 @@
+import json
+from itertools import chain
+from pathlib import Path
+from typing import Any, Generator, Iterable
+
+import pydantic
+
+from cachi2.core.errors import PackageRejected
+from cachi2.core.models.input import YarnClassicPackageInput
+from cachi2.core.rooted_path import PathOutsideRoot, RootedPath
+
+
+class Workspace(pydantic.BaseModel):
+    """Workspace model."""
+
+    path: Path  # path to a workspace.
```

Comment:
There's missing a validator for the path if it even exists as part of the project.

Reply:
Validation will be happening elsewhere (namely when we attempt to read package.json to create an instance). A validator here would introduce coupling with filesystem.